File: ExtraWork/test_diary/test_entry.py
import unittest
from ExtraWork.diaryapp.entry import Entry

class TestEntry(unittest.TestCase):

    def test_my_id_number_return_error_message_if_negative(self):
        with self.assertRaises(ValueError):
            Entry(-1, "Emmanuel", "a guy")

    def test_my_entry_return_error_if_title_is_null(self):
        with self.assertRaises(ValueError):
            Entry(1, None, "a guy")

    def test_my_entry_get_title_function(self):
        entry_saver = Entry(1, "Emmanuel", "a guy")
        self.assertEqual(entry_saver.title, "Emmanuel")

    def test_my_entry_get_id_function(self):
        entry_saver = Entry(1, "Emmanuel", "a guy")
        self.assertEqual(entry_saver.id, 1)

File: testDiary/TestEntry.java
import diaryApp.Entry;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class TestEntry{
    @Test
    public void testMyIdNumberReturnErrorMessageIfNegative(){
        assertThrows(IllegalArgumentException.class,() -> {
            Entry entry = new Entry(-1,"Emmanuel","a guy");},
                "id cannot be negative");
    }

    @Test
    public void testMyEntryReturnErrorIfTitleIsNull() {
        assertThrows(
                IllegalArgumentException.class,
                () -> new Entry(1, null, "a guy"),
                "Expected an IllegalArgumentException when title is null"
        );
    }


    @Test
    public void testMyEntryGetIdFunction(){
        Entry entry = new Entry(1,"Emmanuel","a guy","12/2/2020");
        assertEquals("Emmanuel",entry.getTitle());
    }
}